src/class-wpml-elementor-translatable-nodes.php

```diff
@@ -49,28 +31,26 @@ public function get( $node_id, $element ) {
 		foreach ( $this->nodes_to_translate as $node_type => $node_data ) {
 			if ( $this->conditions_ok( $node_data, $element ) ) {
 				foreach ( $node_data['fields'] as $key => $field ) {
-					$field_key = $field['field'];
+					$field_key    = $field['field'];
+					$string_value = null;
 
-					if ( is_numeric( $key ) && isset( $element[ $this->settings_field ][ $field_key ] ) && trim( $element[ $this->settings_field ][ $field_key ] ) ) {
-						$string    = new WPML_PB_String(
-							$element[ $this->settings_field ][ $field_key ],
-							$this->get_string_name( $node_id, $field, $element ),
-							$field['type'],
-							$field['editor_type'],
-							$this->get_wrap_tag( $element )
-						);
-						$strings[] = $string;
-					} else if ( isset( $element[ $this->settings_field ][ $key ][ $field_key ] ) && trim( $element[ $this->settings_field ][ $key ][ $field_key ] ) ) {
-						$string    = new WPML_PB_String(
-							$element[ $this->settings_field ][ $key ][ $field_key ],
+					if ( isset( $element[ self::SETTINGS_FIELD ][ $field_key ] ) && is_string( $element[ self::SETTINGS_FIELD ][ $field_key ] ) && trim( $element[ self::SETTINGS_FIELD ][ $field_key ] ) ) {
+						$string_value = $element[ self::SETTINGS_FIELD ][ $field_key ];
+					} elseif ( isset( $element[ self::SETTINGS_FIELD ][ $key ][ $field_key ] ) && trim( $element[ self::SETTINGS_FIELD ][ $key ][ $field_key ] ) ) {
```

Comment:
@strategio similar to the previous comment, you can extract this into a private method: it will help readability.

Also, consider what Bruce suggested some time ago (see https://github.com/tightenco/collect which allows you to use https://laravel.com/docs/5.8/collections).

Reply:
@andreasciamanna, I already looked at this library but I prefer to start using it on a new code.